docs/code_style.md

```diff
@@ -114,7 +114,8 @@ Bad:
 
 ### P4 RAII lock type usage
 
-Use `std::scoped_lock` for a common case where a single mutex needs to be locked immediately and unlocked at the end of the scope. Use other lock types sparingly if other features are needed (e.g. multiple mutexes, deferred locking etc).
+Use `std::scoped_lock` for a common case where a single mutex needs to be locked immediately and unlocked at the end of the scope. Do not use `std::lock_guard`.  
```

Comment:
I guess the difference is this:

![image](https://github.com/user-attachments/assets/83a0fbce-5b00-479f-9948-653fab257a39)